NDTensors/src/abstractarray/generic_array_constructors.jl

```diff
@@ -0,0 +1,36 @@
+using .TypeParameterAccessors:
+  unwrap_array_type, specify_default_type_parameters, type_parameter
+
+## Warning to use these functions it is necessary to define `TypeParameterAccessors.position(::Type{<:YourArrayType}, ::typeof(ndims)))`
+# Implementation, catches if `ndims(arraytype) != length(dims)`.
+## TODO convert ndims to `type_parameter(::, typeof(ndims))`
+function generic_randn(arraytype::Type{<:AbstractArray}, dims...; rng=Random.default_rng())
+  arraytype_specified = specify_type_parameter(
+    unwrap_array_type(arraytype), ndims, length(dims)
+  )
+  arraytype_specified = specify_default_type_parameters(arraytype_specified)
+  @assert length(dims) == ndims(arraytype_specified)
+  data = similar(arraytype_specified, dims...)
+  return randn!(rng, data)
+end
+
+function generic_randn(
+  arraytype::Type{<:AbstractArray}, dims::Tuple; rng=Random.default_rng()
+)
+  return generic_randn(arraytype, dims...; rng=rng)
```

Comment:
```suggestion
  return generic_randn(arraytype, dims...; rng)
```
I prefer that style when it can be used.

Reply:
Updated, thanks!